rosbag2_cpp/include/rosbag2_cpp/storage_options.hpp

```diff
@@ -26,11 +26,14 @@ struct StorageOptions
   std::string uri;
   std::string storage_id;
 
-  /**
-   * The maximum size a bagfile can be, in bytes, before it is split.
-   * A value of 0 indicates that bagfile splitting will not be used.
-   */
+  // The maximum size a bagfile can be, in bytes, before it is split.
+  // A value of 0 indicates that bagfile splitting will not be used.
   uint64_t max_bagfile_size;
+
+  // The chunk size indiciates how many messages should be hold in cache
+  // before these being written to disk.
+  // Defaults to 1, and effectively disables the caching.
+  size_t chunk_size = 1;
```

Comment:
Is there a reason why `size_t` is used instead of `uint64_t`?
Also what happens if `0` is used?

Reply:
I know outside of this PR, but why can't the `max_bagfile_size` field above not be similarly defaulted to `0`?

Reply:
IIRC not specifying a value for an int will default it to 0.
The code is more readable with it explicitly initialized to 0, so I think there should be a follow-up PR default initializing `max_bagfile_size` to 0.